kube-core/src/version.rs

```diff
@@ -71,7 +131,7 @@ impl Version {
 
 impl Ord for Version {
```

Comment:
This is confusing.. Does this mean that `Stable(1) > Stable(2)`?  Or are the double negatives supposed to cancel out? In that case, what do we gain compared to just deriving `Ord` for `Version`?

Reply:
Currently, `Stable(2) > Stable(1)`. A property we didn't have before, but should have before making it public.

If we simply derive `Ord`, say with this:

```rust
#[derive(PartialEq, Eq, Debug, Clone, PartialOrd, Ord)]
pub enum Version {
    Other(String),
    Alpha(u32, Option<u32>),
    Beta(u32, Option<u32>),
    Stable(u32),
}
```

(reordering because enum order determines ord), then it's almost right, but we get the wrong answer when sorting nonconformant/other versions:

```
  left: `["v10", "v2", "v1", "v11beta2", "v10beta3", "v3beta1", "v12alpha1", "v11alpha2", "foo10", "foo1"]`, <- got
 right: `["v10", "v2", "v1", "v11beta2", "v10beta3", "v3beta1", "v12alpha1", "v11alpha2", "foo1", "foo10"]`', <- wanted src/version.rs:19:1
```

because kubernetes says to sort nonconformant versions alphabetically and explicitly lists this example

Reply:
I have managed to reduce the number of `Reverse` uses (now only on the nonconformant ones) by relying on derived `Ord` for the sort key - however, we cannot derive `Ord` on the `Version` itself as one reverse is ultimately necessary.